solr/parser.go

```diff
@@ -1,22 +1,52 @@
 package solr
 
 import (
+	"encoding/json"
 	"fmt"
 )
 
+func bytes2json(data *[]byte) (map[string]interface{}, error) {
+	var jsonData interface{}
+
+	err := json.Unmarshal(*data, &jsonData)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return jsonData.(map[string]interface{}), nil
+}
+
 // ResultParser is interface for parsing result from response.
 // The idea here is that application have possibility to parse.
 // Or defined own parser with internal data structure to suite
 // application's need
 type ResultParser interface {
-	Parse(response *SolrResponse) (*SolrResult, error)
+	Parse(resp *[]byte) (*SolrResult, error)
+}
+
+type FireworkResultParser struct {
```

Comment:
It is just name indicating that it is faster than standard parser. We could
as well call it PartialResultParser.

On Wed, 13 Jan 2016 00:19 Nguyen Van Nhu <notifications@github.com> wrote:

> In solr/parser.go
> <https://github.com/vanng822/go-solr/pull/36#discussion_r49484141>:
>
> >  // ResultParser is interface for parsing result from response.
> >  // The idea here is that application have possibility to parse.
> >  // Or defined own parser with internal data structure to suite
> >  // application's need
> >  type ResultParser interface {
> > -	Parse(response *SolrResponse) (*SolrResult, error)
> > +	Parse(resp *[]byte) (*SolrResult, error)
> > +}
> > +
> > +type FireworkResultParser struct {
>
> Just want to know for history @djinn <https://github.com/djinn> In case
> some one ask in the future why we choose this name
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/vanng822/go-solr/pull/36/files#r49484141>.
>